app/src/main/java/com/example/countryinfo/TabletCountryDetailsFragment.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.countryinfo
+
+import android.graphics.drawable.PictureDrawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.net.toUri
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.github.twocoffeesoneteam.glidetovectoryou.GlideApp
+import kotlinx.android.synthetic.main.tablet_country_details_fragment.*
+import kotlinx.android.synthetic.main.tablet_country_details_fragment.view.*
+import javax.inject.Inject
+
+class TabletCountryDetailsFragment : Fragment() {
+
+    companion object {
+        private const val COUNTRY_ID = "country_id"
+        fun newInstance(countryId: String) = TabletCountryDetailsFragment().apply {
+            arguments = bundleOf(COUNTRY_ID to countryId)
+        }
+    }
+
+    private lateinit var rootView: View
+
+    @Inject
+    lateinit var viewModelFactory: CountryDetailsViewModelFactory
+    private val viewModel: CountryDetailsViewModel by viewModels { viewModelFactory }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        rootView = inflater.inflate(R.layout.tablet_country_details_fragment, container, false)
+        (activity?.application as CountriesApplication).countriesComponent.inject(this)
+        return rootView
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        val id = arguments?.get(COUNTRY_ID) as? String
+        viewModel.countryLiveData.observe(viewLifecycleOwner, androidx.lifecycle.Observer {
+            handleState(it)
+        })
+        if (id != null) {
+            viewModel.countryClicked(id)
+        }
+    }
+
+    private fun handleState(countryDetailsViewState: CountryDetailsViewState) {
+        when (countryDetailsViewState) {
+            is CountryDetailsViewState.Default -> {
+                tabletCountryNameTextView.text = countryDetailsViewState.countryDetails.name
```

Comment:
We can leverage Kotlin's `with` function to reduce verbosity
```suggestion
                with(countryDetailsViewState.countryDetails) {
                    tabletCountryNameTextView.text = name
                    .......
                    ......
                }                
```

Reply:
Done.